### ISSUE TITLE
Support ES6 and TypeScript

### ISSUE THREAD

**Issue by voxpelli** [dependency-check-team/dependency-check#131]
As a follow-up to #128:

By default, support `.mjs` and `.ts` using:

* https://github.com/dependents/node-detective-es6
* https://github.com/pahen/detective-typescript

**Comment by fatso83** [comment]
Having working support for TSX is usually also a given when having TypeScript support, which is not straight forward using detective-typescript, so worth having a look at this for a fix: https://github.com/dependency-check-team/dependency-check/issues/112#issuecomment-768251307

**Comment by fatso83** [comment]
Yeah, waiting for all modules to reach some consensus on API design was futile, so it was easier just to take this approach. To avoid any issue of having multiple versions of `detective-typescript` in future updates, I could of course make it into a `peerDependency`, since it's more than likely both will be present in most cases.
